crates/rpc/rpc/src/eth/core.rs

```diff
@@ -129,8 +129,7 @@ where
     Self: Send + Sync,
 {
     type Error = EthApiError;
-    // todo: replace with alloy_network::Ethereum
-    type NetworkTypes = AnyNetwork;
+    type NetworkTypes = Ethereum;
```

Comment:
@emhane Seems like we need on alloy side the `WithOtherFields` wrapper around:
- `TransactionRequest`
- `TransactionResponse`
- `BlockResponse`

in `alloy_network::Ethereum` to match what we had with `alloy_network::AnyNetwork` or if we don't want this we should modify some function definitions like

https://github.com/paradigmxyz/reth/blob/dba6c012955c3e84efde8b2d6eca97c481226588/crates/rpc/rpc-eth-types/src/receipt.rs#L121-L125

which seems not the good option to me, what do you think?

Reply:
> Seems like we need on alloy side the WithOtherFields wrapper around:

TransactionRequest
TransactionResponse
BlockResponse
in alloy_network::Ethereum

hard no on this, this would render the purpose of Network entirely useless

Reply:
I understand, then we need to modify quite a lot of function definitions like this 

```rust
  /// Builds a receipt response from the base response body, and any set additional fields.
    pub fn build(self) -> AnyTransactionReceipt {
        let Self { base, other } = self;
        WithOtherFields { inner: base, other }
    }
```

becoming something like
```rust
  /// Builds a receipt response from the base response body, and any set additional fields.
    pub fn build(self) -> TransactionReceipt {
        let Self { base, other } = self;
        base
    }
```

is it ok to do that to a bunch of functions? Is it the right approach to take?

Reply:
yeah I think the ReceiptBuilder type is tmp anyway because this is something that is entirely network specific.

I don't have a solution for this yet, because due to how the default helper traits work, this is a bit tricky I guess

Reply:
also not that critical atm, because from server POV the Any* types work okay

Reply:
let us mark this as blocked maybe atm?

Reply:
would be v helpful if you created subtasks to the issue for all the blockers you encounter, to gradually unblock this @tcoratger. for example modifying the receipt builder is one subtask.

Reply:
Done https://github.com/paradigmxyz/reth/issues/11219